api/python/t4/snapshots.py

```diff
@@ -554,18 +564,63 @@ def top_hash(self):
             self._top_hash()
         return self._meta['top_hash']
 
-    def materialize(self, path, name=None):
+    def push(self, name, path):
         """
         Copies objects to path, then creates a new package that points to those objects.
-
         Copies each object in this package to path according to logical key structure,
         then adds to the registry a serialized version of this package
         with physical_keys that point to the new copies.
+        Args:
+            path: where to copy the objects in the package
+            name: name for package in registry
+        Returns:
+            A new package that points to the copied objects
+        """
+        if not path.startswith('s3://'):
+            raise NotImplementedError
+        if not name:
+            # todo: handle where to put data for unnamed remote packages
+            raise NotImplementedError
+        pkg = self._materialize('{}/{}'.format(path.strip("/"), name))
+
+        with tempfile.NamedTemporaryFile() as manifest:
+            pkg.dump(manifest)
+            manifest.flush()
+            _copy_file(
+                _fix_url(manifest.name),
```

Comment:
I meant it to be used on user input where we want to support both URIs and file paths - and possibly use heuristics to guess which one it is. When we know it's a file path, I'd rather use `Path(manifest.name).resolve().as_uri()` - or change the other helpers to make it less verbose.